Makefile

```diff
@@ -28,74 +27,78 @@ endif
 # Some will have dedicated targets to make it easier to type, for example
 # "apiserver" instead of "bin/apiserver".
 #########################################################################
-build: init \
+build: .init \
        $(BINDIR)/controller $(BINDIR)/registry $(BINDIR)/k8s-broker \
        $(BINDIR)/service-catalog $(BINDIR)/user-broker \
        $(BINDIR)/apiserver
 
 controller: $(BINDIR)/controller
-$(BINDIR)/controller: pkg/controller/catalog
-	$(DOCKER) $(GO_BUILD) -o $@ $(SC_PKG)/$^
+$(BINDIR)/controller: pkg/controller/catalog $(shell find pkg/controller/catalog -type f)
+	$(DOCKER) $(GO_BUILD) -o $@ $(SC_PKG)/pkg/controller/catalog
 
 registry: $(BINDIR)/registry
-$(BINDIR)/registry: contrib/registry
-	$(DOCKER) $(GO_BUILD) -o $@ $(SC_PKG)/$^
+$(BINDIR)/registry: contrib/registry $(shell find contrib/registry -type f)
+	$(DOCKER) $(GO_BUILD) -o $@ $(SC_PKG)/contrib/registry
 
-$(BINDIR)/k8s-broker: contrib/broker/k8s
-	$(DOCKER) $(GO_BUILD) -o $@ $(SC_PKG)/$^
+$(BINDIR)/k8s-broker: contrib/broker/k8s $(shell find contrib/broker/k8s -type f)
```

Comment:
this doesn't seem correct as it doesn't take into effect all dependencies:
https://github.com/kubernetes-incubator/service-catalog/blob/master/contrib/broker/k8s/broker.go#L26

I'm quite sure other rules have the same problem.